fcsparser/api.py

```diff
@@ -58,12 +59,14 @@ class FCSParser(object):
         self.channel_names_alternate holds the alternate names of the channels
     """
 
-    def __init__(self, path, read_data=True, channel_naming='$PnS'):
+    def __init__(self, path, dataset_start=0, read_data=True, channel_naming='$PnS'):
```

Comment:
That is a good idea. I will work on it.

----- Original Message -----
From: "Eugene Yurtsev" <notifications@github.com>
To: "eyurtsev/fcsparser" <fcsparser@noreply.github.com>
Cc: "meihuisu" <mei@isi.edu>
Sent: Sunday, March 20, 2016 6:33:41 PM
Subject: Re: [fcsparser] Allow $NEXTDATA to be $ENDDATA+1 (#2)

> @@ -58,12 +59,14 @@ class FCSParser(object):
>          self.channel_names_alternate holds the alternate names of the channels
>      """
>  
> -    def __init__(self, path, read_data=True, channel_naming='$PnS'):
> +    def __init__(self, path, dataset_start=0, read_data=True, channel_naming='$PnS'):

dataset_start is not very convenient for a user to specify since it requires the user to extract the meta parameters and then feed the meta parameter back into the parser.

Instead of specifying an offset in terms of bytes in a binary file, how about we specify an offset in terms of dataset blocks?

For example, to get the first dataset a user would provide:

```
dataset_num = 0
```

To get the second dataset:
```
dataset_num = 1
```

and so on...





---
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/eyurtsev/fcsparser/pull/2/files/cebd3b867784c6c898716411f83c9efb78fe448f#r56777362